### PR TITLE
Encryption id fix #issue34

### PR DESCRIPTION
**Pull request by Guthix2827**
following this issue: https://github.com/OmarElGabry/miniPHP/issues/34#issuecomment-723137154
that's a solution I have implement long time ago and its stable and works fine at one of my production systems
@OmarElGabry

### CODE REVIEW

File: app/core/Encryption.php [hunk 1]
```diff
@@ -255,4 +226,157 @@ private static function hashEquals($hmac, $compare){
...
+	 * @license   http://www.opensource.org/licenses/bsd-license.php New BSD Licence
+	 * @version   SVN: Release: $Id: alphaID.inc.php 344 2009-06-10 17:43:59Z kevin $
+	 * @link	  http://kevin.vanzonneveld.net/
+	 *
+	 * @param mixed   $in	  String or long input to translate
+	 * @param boolean $to_num  Reverses translation when true
+	 * @param mixed   $pad_up  Number or boolean padds the result up to a specified length
+	 * @param string  $pass_key Supplying a password makes it harder to calculate the original ID
+	 *
+	 * @return mixed string or long
```
Comment (OmarElGabry) [inline app/core/Encryption.php:317]: are you sure this can be long? It seems it is only string?

File: app/core/Encryption.php [hunk 2]
```diff
@@ -255,4 +226,157 @@ private static function hashEquals($hmac, $compare){
...
+	 * @author	Kevin van Zonneveld <kevin@vanzonneveld.net>
+	 * @author	Simon Franz
+	 * @author	Deadfish
+	 * @author  SK83RJOSH
+	 * @copyright 2008 Kevin van Zonneveld (http://kevin.vanzonneveld.net)
+	 * @license   http://www.opensource.org/licenses/bsd-license.php New BSD Licence
+	 * @version   SVN: Release: $Id: alphaID.inc.php 344 2009-06-10 17:43:59Z kevin $
+	 * @link	  http://kevin.vanzonneveld.net/
+	 *
+	 * @param mixed   $in	  String or long input to translate
```
Comment (OmarElGabry) [inline app/core/Encryption.php:312]: I am a bit confused to whether the function really takes in string or long, or just string? shouldn't we cast it?

File: app/core/Encryption.php [hunk 3]
```diff
@@ -255,4 +226,157 @@ private static function hashEquals($hmac, $compare){
...
+	 * // 2188847690240 => SpQXn7Cb
+	 * // SpQXn7Cb => 2188847690240
+	 * // aaab => 238328
+	 *
+	 * </code>
+	 *
+	 * @author	Kevin van Zonneveld <kevin@vanzonneveld.net>
+	 * @author	Simon Franz
+	 * @author	Deadfish
+	 * @author  SK83RJOSH
```
Comment (OmarElgabry) [inline app/core/Encryption.php]: fix spacing

**Review by OmarElgabry — CHANGES_REQUESTED** [review]
I don't see a link to the source code:

https://kvz.io/create-short-ids-with-php-like-youtube-or-tinyurl.html

**Comment by Guthix2827** [comment]
@OmarElGabry

**Pull request merged**